src/ucs/datastruct/strided_alloc.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2018.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_STRIDED_ALLOC_H_
+#define UCS_STRIDED_ALLOC_H_
+
+#include "queue_types.h"
+
+#include <ucs/type/status.h>
+#include <ucs/sys/compiler_def.h>
+#include <ucs/sys/math.h>
+#include <stddef.h>
+
+
+BEGIN_C_DECLS
+
+
+/* the distance between allocated elements */
+#define UCS_STRIDED_ALLOC_STRIDE (128 * UCS_KBYTE)
+
+
+/**
+ * Get a pointer to another element in the strided object
+ *
+ * Example with stride_count=3:
+ *
+ * chunk
+ * start -+
+ *        |
+ *        \/
+ *        |
+ *        +--------+--   ...   --+--------+--   ...   --+--------+
+ *        | stride |             | stride |             | stride |
+ * obj0:  | elem 0 | <- 128kB -> | elem 1 | <- 128kB -> | elem 2 |
+ *        | (base) |             |        |             |        |
+ *        +--------+--   ...   --+--------+--   ...   --+--------+
+ *                 +--------+--  ...    --+--------+--  ...    --+--------+
+ *                 | stride |             | stride |             | stride |
+ * obj1:           | elem 0 | <- 128kB -> | elem 1 | <- 128kB -> | elem 2 |
+ *                 | (base) |             |        |             |        |
+ *                 +--------+--  ...     -+--------+--  ...    --+--------+
+ *                          +--------+--  ...    --+--------+--   ...   --+--------+
+ *                          | stride |             | stride |             | stride |
+ * obj2:                    | elem 0 | <- 128kB -> | elem 1 | <- 128kB -> | elem 2 |
+ *                          | (base) |             |        |             |        |
+ *                          +--------+--  ...     -+--------+--   ...   --+--------+
+ *
+ * ...
+ *
+ * @param _elem        Pointer to the current element
+ * @param _stride_idx  Stride index of the current element
+ * @param _wanted_idx  Stride index of the desired element
+ *
+ * @return Pointer to the desired element
+ */
+#define ucs_strided_alloc_step(_elem, _stride_idx, _wanted_idx) \
```

Comment:
name is a bit confusing and `_strided_idx` looks not needed because we have to pass base `elem` anyway. Maybe rename to something like `ucs_strided_elem_get(_elem, _stride)` or add one more wrapper?

Reply:
its also used to get the base elem from a stride elem

Reply:
Do you mean like this: `ucs_strided_alloc_step(elem2, 2, 0)` ?
Any way i'm voting for `ucs_strided_elem_get(_elem, _elem_idx, _stride_idx)`

Reply:
yes, `ucs_strided_alloc_step(elem2, 2, 0)`
will rename